contracts/apps/20-transfer/ICS20TransferERC20.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "./ICS20Transfer.sol";
+import "../../core/25-handler/IBCHandler.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/presets/ERC20PresetMinterPauser.sol";
+
+// An ICS20 implementation that maps sink denoms to ERC20 contracts that's deployed and managed by this contract.
+//
+// Source denom is interpreted as ERC20 contract address in hex with the 0x prefix.
+contract ICS20TransferERC20 is ICS20Transfer {
+    // Map sink denom to ERC20.
+    mapping(string => ERC20PresetMinterPauser) public denomTokenContract;
+
+    constructor(IBCHandler ibcHandler_) ICS20Transfer(ibcHandler_) {
+    }
+
+    function _transferFrom(address sender, address receiver, string memory denom, uint256 amount)
+        internal
+        override
+        returns (bool)
+    {
+        IERC20 tokenContract = IERC20(parseAddr(denom));
+        // transferFrom returns a bool but it may also revert.
+        try tokenContract.transferFrom(sender, receiver, amount) returns (bool succeeded) {
+            return succeeded;
+        } catch (bytes memory) {
+            return false;
+        }
+    }
+
+    function _mint(address account, string memory denom, uint256 amount) internal override returns (bool) {
+        // Deploy an ERC20 contract for each (sink zone) denom seen.
+        if (address(denomTokenContract[denom]) == address(0)) {
+            string memory name = string.concat("IBC/", hexEncode(abi.encodePacked(sha256(bytes(denom)))));
+            denomTokenContract[denom] = new ERC20PresetMinterPauser(name, "");
```

Comment:
how to associate this non-human-readable token name with a human-readable token name, like ETC, BTC, etc.

Reply:
It's a concern for UI/explorer etc. Cosmos uses https://github.com/cosmos/chain-registry.

Reply:
Or the allowlist version allows setting custom token name, symbol and decimal.

Reply:
Why do not set symbol for the ERC20?

Reply:
What should the symbol be?

Reply:
I am no sure if we can use `denom` as the symbol, but use an empty string is strange

Reply:
> I am no sure if we can use `denom` as the symbol, but use an empty string is strange

it's not an empty string, but just an encoded string which isn't recognized for human-beings

Reply:
I understand the first arg.

``` solidity
            denomTokenContract[denom] = new ERC20PresetMinterPauser(name, "");
```

I am discussing the second one